pkg/controller/migration/convert/network.go

```diff
@@ -110,10 +147,123 @@ func handleCalicoCNI(c *components, install *Installation) error {
 	if c.calicoCNIConfig.FeatureControl.IPAddrsNoIpam {
 		return ErrIncompatibleCluster{"IpAddrsNoIpam not supported"}
 	}
+	if c.calicoCNIConfig.ContainerSettings.AllowIPForwarding {
+		return ErrIncompatibleCluster{"AllowIPForwarding not supported"}
+	}
+
+	return nil
+}
+
+func subhandleCalicoIPAM(c *components, install *Installation) error {
```

Comment:
minor suggestion - could probably _just_ pass in `c.calicoCNIConfig.IPAM` instead of the whole `*components` struct.

Reply:
Hmm, I kind of moved some other stuff into the subhandlers, see what you think about that.
I think I like keeping the (sub)handlers all with the same interface, but I'm not sure if that's a good reason.
... Looks like I might have broke something.

Reply:
it might help cut down on all the nil checks that you have to do if these just accepted the bits they're processing :shrug: